GAE/src/com/gallatinsystems/survey/domain/WebForm.java

```diff
@@ -25,17 +25,26 @@
 
     public static Set<String> unsupportedQuestionTypes(){
         Set<String> unsupportedTypes = new HashSet<String>();
-        unsupportedTypes.add(Question.Type.CASCADE.toString());
         unsupportedTypes.add(Question.Type.GEOSHAPE.toString());
         unsupportedTypes.add(Question.Type.SIGNATURE.toString());
         unsupportedTypes.add(Question.Type.CADDISFLY.toString());
         return unsupportedTypes;
     }
 
-    public static boolean validWebForm(final List<Question> questions){
-        
-        List<Question> validQuestions = questions.stream().filter(i -> !unsupportedQuestionTypes().contains(i.getType().toString())).collect(Collectors.toList());
+    public static boolean validQuestionGroups(final Survey survey){
+        return survey.getQuestionGroupMap().values().stream().filter(i -> i.getRepeatable().booleanValue()).collect(Collectors.toList()).size() == 0;
```

Comment:
Why do we need an explicit `boolean` primitive? .... using `i.getRepeatable()` does the trick